core/test.ts

```diff
@@ -24,8 +24,14 @@ export class Test {
     return this;
   }
 
-  public dataset(datasetToTest: string) {
-    this.datasetToTest = datasetToTest;
+  public dataset(ref: Resolvable) {
+    const target2Resolvable = (t: dataform.ITarget) => {
```

Comment:
We need to call `target()` to add the suffix. If not, then the following test will not pass: `bigquery using v2 language compiles with suffix "suffix"` for the file `biguquery_language_v2/definitions/example_test_case_fq_ref`.

Notice that the ability to use FQ refs on tests is a functionality that nobody asked for. If it is controvesial, I am happy removing it but what will happen is that you will not be able to reference a FQ `dataset` on a Test like this:

```
config {
    type: "test",
    dataset: {schema: "df_integration_test", name: "example_table"}
}
```

This only occurs in the case where there is a **SUFFIX**.  We need to call the `target()` function to add the suffix.

I have changed the code and moved the `target()` resolution inside the compile phase so that the code is cleaner.

This is arguably a very strange case because two things must happen for this code to make sense: (1) User must refer a Fully-Qualified dataset on a Test; (2) User must use a suffix. But still, it can happen!

Reply:
Ahhhhh, thank you for explaining, I definitely didn't consider the case for schema suffixes!